.github/workflows/ci.yml

```diff
@@ -0,0 +1,39 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  windows-tests:
+    runs-on: windows-latest
```

Comment:
```suggestion
  run-tests:
    runs-on: [ubuntu-latest, macos-latest, windows-latest]
```

Reply:
currently only test on ubuntu as the other runners are not available in a timely fashion

Reply:
What do you mean?

Reply:
https://docs.github.com/en/actions/using-jobs/choosing-the-runner-for-a-job#overview

fourth bullet: "If you specify an array of strings or variables, your workflow will execute on any runner that matches all of the specified runs-on values." 
it's unlikely that a single runner exists with all the requested operating systems

I modified the workflow to create a matrix over the supported Python versions and operating systems to address this issue. This change allows your workflow to run on multiple combinations of both Python versions and OSs during each CI job execution.

Reply:
Oh yeah, sorry. That's why I meant. Thanks!